### ISSUE TITLE
A long overdue PyTorch version is here

### ISSUE THREAD

**Issue by voldemortX** [XingangPan/SCNN#125]
@XingangPan 
It is now 2021, SCNN (proposed over 3 years ago) has become **the** classic method for lane detection and still pose a challenge to SOTA methods. But the old torch7 code and matlab evaluation is just not enough for the community anymore. So we have been working on a unified lane detection codebase, and have now re-implemented **a PyTorch version of SCNN** (https://github.com/voldemortX/pytorch-auto-drive) on **6** different backbones (VGG16, ResNet18/34/50/101, ERFNet) and achieved better results than originally reported in the paper (we attribute the improvements mainly to large batch size, better pre-trained weights, larger learning rate with warmup). For instance, we find VGG+SCNN can reach **74.29** F1 score on CULane, almost 3% higher than the paper result. More results are available in our [model zoo](https://github.com/voldemortX/pytorch-auto-drive/blob/master/docs/MODEL_ZOO.md).
Although we could not get the TuSimple competition performance (96.53%) from VGG + SCNN training on the train set alone, we assume that is normal since it was indeed a competition.

This can help researchers who use Python/PyTorch to ease reproduction efforts: #117 #70 #69 #58 #6

We would also be providing visualization tools and fair FPS tests across methods, which can help issues like this: #20

Maybe you could add a link to our PyTorch version in `README.md` to help promote the codebase?

p.s. We use python to map predictions to lanes, there is no need for matlab anymore.

**Comment by XingangPan** [comment]
@voldemortX Thanks for your efforts. Your pytorch-auto-drive code base looks great! It's glad to know that your reproduction obtains even better results. I have added the link in `README.md`.

**Comment by voldemortX** [comment]
Thanks a lot for your support!
